Make it compile and work on Windows

```
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\heed-0.5.0\src\env.rs:5:14
  |
5 | use std::os::unix::io::AsRawFd;
  |              ^^^^ could not find `unix` in `os`

error[E0599]: no method named `as_raw_fd` found for type `std::fs::File` in the current scope
   --> C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\heed-0.5.0\src\env.rs:307:23
    |
307 |         let fd = file.as_raw_fd();
    |                       ^^^^^^^^^ method not found in `std::fs::File`

error: aborting due to 2 previous errors
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,6 +21,9 @@ serde = { version = "1.0.101", optional = true }
 serde_json = { version = "1.0.41", optional = true }
 zerocopy = "0.2.8"
 
+[target.'cfg(windows)'.dependencies]
+url = "2.1.0"
+
 [dev-dependencies]
 serde = { version = "1.0.101", features = ["derive"] }
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,8 +13,8 @@ edition = "2018"
 [dependencies]
 bincode = { version = "1.2.0", optional = true }
 byteorder = { version = "1.3", default-features = false }
-lmdb-rkv-sys = "0.9.5"
 libc = "0.2.62"
+lmdb-rkv-sys = "0.10.0"
 once_cell = "1.2.0"
 page_size = "0.4.1"
 serde = { version = "1.0.101", optional = true }

--- a/examples/all-types-poly.rs
+++ b/examples/all-types-poly.rs
@@ -1,5 +1,6 @@
 use std::error::Error;
 use std::fs;
+use std::path::Path;
 
 use heed::byteorder::BE;
 use heed::types::*;
@@ -8,12 +9,12 @@ use heed::EnvOpenOptions;
 use serde::{Deserialize, Serialize};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    fs::create_dir_all("target/zerocopy-dyn.mdb")?;
+    fs::create_dir_all(Path::new("target").join("zerocopy-dyn.mdb"))?;
 
     let env = EnvOpenOptions::new()
         .map_size(10 * 1024 * 1024 * 1024) // 10GB
         .max_dbs(3000)
-        .open("target/zerocopy-dyn.mdb")?;
+        .open(Path::new("target").join("zerocopy-dyn.mdb"))?;
 
     // you can specify that a database will support some typed key/data
     //

--- a/examples/all-types.rs
+++ b/examples/all-types.rs
@@ -1,5 +1,6 @@
 use std::error::Error;
 use std::fs;
+use std::path::Path;
 
 use heed::byteorder::BE;
 use heed::types::*;
@@ -8,12 +9,12 @@ use heed::{Database, EnvOpenOptions};
 use serde::{Deserialize, Serialize};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    fs::create_dir_all("target/zerocopy.mdb")?;
+    fs::create_dir_all(Path::new("target").join("zerocopy-dyn.mdb"))?;
 
     let env = EnvOpenOptions::new()
         .map_size(10 * 1024 * 1024 * 1024) // 10GB
         .max_dbs(3000)
-        .open("target/zerocopy.mdb")?;
+        .open(Path::new("target").join("zerocopy.mdb"))?;
 
     // you can specify that a database will support some typed key/data
     //

--- a/examples/nested.rs
+++ b/examples/nested.rs
@@ -1,16 +1,17 @@
 use std::error::Error;
 use std::fs;
+use std::path::Path;
 
 use heed::types::*;
 use heed::{Database, EnvOpenOptions};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    fs::create_dir_all("target/zerocopy.mdb")?;
+    fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
 
     let env = EnvOpenOptions::new()
         .map_size(10 * 1024 * 1024 * 1024) // 10GB
         .max_dbs(3000)
-        .open("target/zerocopy.mdb")?;
+        .open(Path::new("target").join("zerocopy.mdb"))?;
 
     // here the key will be an str and the data will be a slice of u8
     let db: Database<Str, ByteSlice> = env.create_database(None)?;

--- a/src/db/polymorph.rs
+++ b/src/db/polymorph.rs
@@ -19,17 +19,18 @@ use crate::*;
 ///
 /// ```
 /// # use std::fs;
+/// # use std::path::Path;
 /// # use heed::EnvOpenOptions;
 /// use heed::PolyDatabase;
 /// use heed::types::*;
 /// use heed::{zerocopy::I64, byteorder::BigEndian};
 ///
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # fs::create_dir_all("target/zerocopy.mdb")?;
+/// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
 /// # let env = EnvOpenOptions::new()
 /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
 /// #     .max_dbs(3000)
-/// #     .open("target/zerocopy.mdb")?;
+/// #     .open(Path::new("target").join("zerocopy.mdb"))?;
 /// type BEI64 = I64<BigEndian>;
 ///
 /// let db: PolyDatabase = env.create_poly_database(Some("big-endian-iter"))?;
@@ -61,17 +62,18 @@ use crate::*;
 ///
 /// ```
 /// # use std::fs;
+/// # use std::path::Path;
 /// # use heed::EnvOpenOptions;
 /// use heed::PolyDatabase;
 /// use heed::types::*;
 /// use heed::{zerocopy::I64, byteorder::BigEndian};
 ///
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # fs::create_dir_all("target/zerocopy.mdb")?;
+/// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
 /// # let env = EnvOpenOptions::new()
 /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
 /// #     .max_dbs(3000)
-/// #     .open("target/zerocopy.mdb")?;
+/// #     .open(Path::new("target").join("zerocopy.mdb"))?;
 /// type BEI64 = I64<BigEndian>;
 ///
 /// let db: PolyDatabase = env.create_poly_database(Some("big-endian-iter"))?;
@@ -118,16 +120,17 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// let db = env.create_poly_database(Some("get-poly-i32"))?;
     ///
     /// let mut wtxn = env.write_txn()?;
@@ -186,17 +189,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("first-poly-i32"))?;
@@ -236,17 +240,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("last-i32"))?;
@@ -282,17 +287,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -335,17 +341,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -380,17 +387,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -423,17 +431,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -481,17 +490,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -561,17 +571,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -654,17 +665,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -720,17 +732,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -796,17 +809,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -860,17 +874,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -928,17 +943,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;
@@ -989,17 +1005,18 @@ impl PolyDatabase {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db = env.create_poly_database(Some("iter-i32"))?;

--- a/src/db/uniform.rs
+++ b/src/db/uniform.rs
@@ -15,17 +15,18 @@ use crate::*;
 ///
 /// ```
 /// # use std::fs;
+/// # use std::path::Path;
 /// # use heed::EnvOpenOptions;
 /// use heed::Database;
 /// use heed::types::*;
 /// use heed::{zerocopy::I64, byteorder::BigEndian};
 ///
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # fs::create_dir_all("target/zerocopy.mdb")?;
+/// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
 /// # let env = EnvOpenOptions::new()
 /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
 /// #     .max_dbs(3000)
-/// #     .open("target/zerocopy.mdb")?;
+/// #     .open(Path::new("target").join("zerocopy.mdb"))?;
 /// type BEI64 = I64<BigEndian>;
 ///
 /// let db: Database<OwnedType<BEI64>, Unit> = env.create_database(Some("big-endian-iter"))?;
@@ -61,17 +62,18 @@ use crate::*;
 ///
 /// ```
 /// # use std::fs;
+/// # use std::path::Path;
 /// # use heed::EnvOpenOptions;
 /// use heed::Database;
 /// use heed::types::*;
 /// use heed::{zerocopy::I64, byteorder::BigEndian};
 ///
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # fs::create_dir_all("target/zerocopy.mdb")?;
+/// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
 /// # let env = EnvOpenOptions::new()
 /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
 /// #     .max_dbs(3000)
-/// #     .open("target/zerocopy.mdb")?;
+/// #     .open(Path::new("target").join("zerocopy.mdb"))?;
 /// type BEI64 = I64<BigEndian>;
 ///
 /// let db: Database<OwnedType<BEI64>, Unit> = env.create_database(Some("big-endian-iter"))?;
@@ -132,16 +134,17 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// let db: Database<Str, OwnedType<i32>> = env.create_database(Some("get-i32"))?;
     ///
     /// let mut wtxn = env.write_txn()?;
@@ -174,17 +177,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("first-i32"))?;
@@ -216,17 +220,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("last-i32"))?;
@@ -254,17 +259,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -295,17 +301,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -336,17 +343,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -375,17 +383,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -429,17 +438,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -480,17 +490,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -544,17 +555,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<Str, OwnedType<BEI32>> = env.create_database(Some("iter-i32"))?;
@@ -595,17 +607,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<Str, OwnedType<BEI32>> = env.create_database(Some("iter-i32"))?;
@@ -656,17 +669,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -698,17 +712,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -750,17 +765,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;
@@ -803,17 +819,18 @@ impl<KC, DC> Database<KC, DC> {
     ///
     /// ```
     /// # use std::fs;
+    /// # use std::path::Path;
     /// # use heed::EnvOpenOptions;
     /// use heed::Database;
     /// use heed::types::*;
     /// use heed::{zerocopy::I32, byteorder::BigEndian};
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// # fs::create_dir_all("target/zerocopy.mdb")?;
+    /// # fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// # let env = EnvOpenOptions::new()
     /// #     .map_size(10 * 1024 * 1024 * 1024) // 10GB
     /// #     .max_dbs(3000)
-    /// #     .open("target/zerocopy.mdb")?;
+    /// #     .open(Path::new("target").join("zerocopy.mdb"))?;
     /// type BEI32 = I32<BigEndian>;
     ///
     /// let db: Database<OwnedType<BEI32>, Str> = env.create_database(Some("iter-i32"))?;

--- a/src/env.rs
+++ b/src/env.rs
@@ -4,7 +4,7 @@ use std::ffi::CString;
 use std::fs::File;
 use std::path::{Path, PathBuf};
 use std::sync::{Arc, Mutex};
-use std::{ptr, sync};
+use std::{io, ptr, sync};
 #[cfg(windows)]
 use std::ffi::OsStr;
 #[cfg(unix)]
@@ -16,8 +16,22 @@ use crate::{Database, Error, PolyDatabase, Result, RoTxn, RwTxn};
 use lmdb_sys as ffi;
 use once_cell::sync::OnceCell;
 
-
 static OPENED_ENV: OnceCell<Mutex<HashMap<PathBuf, Env>>> = OnceCell::new();
+
+// Thanks to the mozilla/rkv project
+// Workaround the UNC path on Windows, see https://github.com/rust-lang/rust/issues/42869.
+// Otherwise, `Env::from_env()` will panic with error_no(123).
+#[cfg(not(windows))]
+fn canonicalize_path(path: &Path) -> io::Result<PathBuf> {
+    path.canonicalize()
+}
+
+#[cfg(windows)]
+fn canonicalize_path(path: &Path) -> io::Result<PathBuf> {
+    let canonical = path.canonicalize()?;
+    let url = url::Url::from_file_path(&canonical).map_err(|_e| io::Error::new(io::ErrorKind::Other, "URL passing error"))?;
+    url.to_file_path().map_err(|_e| io::Error::new(io::ErrorKind::Other, "path canonicalization error"))
+}
 
 #[cfg(windows)]
 /// Adding a 'missing' trait from windows OsStrExt
@@ -122,7 +136,7 @@ impl EnvOpenOptions {
 
     pub fn open<P: AsRef<Path>>(&self, path: P) -> Result<Env> {
         let path = path.as_ref();
-        let path = path.canonicalize()?;
+        let path = canonicalize_path(path)?;
 
         let mutex = OPENED_ENV.get_or_init(Mutex::default);
         let mut lock = mutex.lock().unwrap();

--- a/src/env.rs
+++ b/src/env.rs
@@ -2,7 +2,6 @@ use std::any::TypeId;
 use std::collections::hash_map::{Entry, HashMap};
 use std::ffi::CString;
 use std::fs::File;
-use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
 use std::sync::{Arc, Mutex};
 use std::{ptr, sync};
@@ -304,7 +303,10 @@ impl Env {
         RoTxn::new(self.0.env)
     }
 
+    #[cfg(not(windows))]
     pub fn copy_to_path<P: AsRef<Path>>(&self, path: P, option: CompactionOption) -> Result<File> {
+        use std::os::unix::io::AsRawFd;
+
         let flags = if let CompactionOption::Enabled = option {
             ffi::MDB_CP_COMPACT
         } else {
@@ -315,6 +317,24 @@ impl Env {
         let fd = file.as_raw_fd();
 
         unsafe { lmdb_result(ffi::mdb_env_copyfd2(self.0.env, fd, flags))? }
+
+        Ok(file)
+    }
+
+    #[cfg(windows)]
+    pub fn copy_to_path<P: AsRef<Path>>(&self, path: P, option: CompactionOption) -> Result<File> {
+        use std::os::windows::io::AsRawHandle;
+
+        let flags = if let CompactionOption::Enabled = option {
+            ffi::MDB_CP_COMPACT
+        } else {
+            0
+        };
+
+        let file = File::create(path)?;
+        let handle = file.as_raw_handle();
+
+        unsafe { lmdb_result(ffi::mdb_env_copyfd2(self.0.env, handle, flags))? }
 
         Ok(file)
     }

--- a/src/env.rs
+++ b/src/env.rs
@@ -59,18 +59,19 @@ impl EnvOpenOptions {
     /// Set one or more LMDB flags (see http://www.lmdb.tech/doc/group__mdb__env.html).
     /// ```
     /// use std::fs;
+    /// use std::path::Path;
     /// use heed::{EnvOpenOptions, Database};
     /// use heed::types::*;
     /// use heed::flags::Flags;
     ///
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-    /// fs::create_dir_all("target/zerocopy.mdb")?;
+    /// fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
     /// let mut env_builder = EnvOpenOptions::new();
     /// unsafe {
     ///     env_builder.flag(Flags::MdbNoSync);
     ///     env_builder.flag(Flags::MdbNoMetaSync);
     /// }
-    /// let env = env_builder.open("target/zerocopy.mdb")?;
+    /// let env = env_builder.open(Path::new("target").join("zerocopy.mdb"))?;
     ///
     /// // we will open the default unamed database
     /// let db: Database<Str, OwnedType<i32>> = env.create_database(None)?;

--- a/src/env.rs
+++ b/src/env.rs
@@ -5,6 +5,10 @@ use std::fs::File;
 use std::path::{Path, PathBuf};
 use std::sync::{Arc, Mutex};
 use std::{ptr, sync};
+#[cfg(windows)]
+use std::ffi::OsStr;
+#[cfg(unix)]
+use std::os::unix::ffi::OsStrExt;
 
 use crate::flags::Flags;
 use crate::lmdb_error::lmdb_result;
@@ -12,7 +16,20 @@ use crate::{Database, Error, PolyDatabase, Result, RoTxn, RwTxn};
 use lmdb_sys as ffi;
 use once_cell::sync::OnceCell;
 
+
 static OPENED_ENV: OnceCell<Mutex<HashMap<PathBuf, Env>>> = OnceCell::new();
+
+#[cfg(windows)]
+/// Adding a 'missing' trait from windows OsStrExt
+trait OsStrExtLmdb {
+    fn as_bytes(&self) -> &[u8];
+}
+#[cfg(windows)]
+impl OsStrExtLmdb for OsStr {
+    fn as_bytes(&self) -> &[u8] {
+        &self.to_str().unwrap().as_bytes()
+    }
+}
 
 #[derive(Clone, Debug)]
 pub struct EnvOpenOptions {
@@ -114,8 +131,7 @@ impl EnvOpenOptions {
             Entry::Occupied(entry) => Ok(entry.get().clone()),
             Entry::Vacant(entry) => {
                 let path = entry.key();
-                let path = path.to_string_lossy();
-                let path = CString::new(path.as_bytes()).unwrap();
+                let path = CString::new(path.as_os_str().as_bytes()).unwrap();
 
                 unsafe {
                     let mut env: *mut ffi::MDB_env = ptr::null_mut();

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -18,12 +18,13 @@
 //!
 //! ```
 //! use std::fs;
+//! use std::path::Path;
 //! use heed::{EnvOpenOptions, Database};
 //! use heed::types::*;
 //!
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
-//! fs::create_dir_all("target/zerocopy.mdb")?;
-//! let env = EnvOpenOptions::new().open("target/zerocopy.mdb")?;
+//! fs::create_dir_all(Path::new("target").join("zerocopy.mdb"))?;
+//! let env = EnvOpenOptions::new().open(Path::new("target").join("zerocopy.mdb"))?;
 //!
 //! // we will open the default unamed database
 //! let db: Database<Str, OwnedType<i32>> = env.create_database(None)?;

